internal/core/service/user_service.go

```diff
@@ -0,0 +1,158 @@
+package service
+
+import (
+	"errors"
+	"github.com/gin-gonic/gin"
+	"github.com/viniciusgferreira/ps-tag-onboarding-go/internal/core/domain/models"
+	"github.com/viniciusgferreira/ps-tag-onboarding-go/internal/core/ports"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.mongodb.org/mongo-driver/mongo"
+	"regexp"
+)
+
+type UserService struct {
+	repo ports.UserRepository
+}
+
+type InvalidUserResponse struct {
+	Name    string   `json:"error"`
+	Details []string `json:"details,omitempty"`
+}
+
+func (r InvalidUserResponse) Error() string {
+	return r.Name
+}
+
+func (r InvalidUserResponse) New(validationErrors []error) InvalidUserResponse {
+	details := make([]string, len(validationErrors))
+	for i, err := range validationErrors {
+		details[i] = err.Error()
+	}
+	return InvalidUserResponse{
+		Name:    "User did not pass validation",
+		Details: details,
+	}
+}
+
```

Comment:
I think it looks better to call this function as newValidationErrors() with no receiver